### ISSUE TITLE
Artifact of chosen sometimes ignores the given aspect and gives you another.

### ISSUE THREAD

**Issue by ConqAra** [CuteDoge01/ArtifactOfChosenMod#2]
Artifact of chosen sometimes ignores taking the aspect away and just gives me another one, most likely because I went to the bazaar? It was fine the first time i went to the bazaar i noticed it kept my aspect but as i left it was back to normal with a new aspect. but the second time i went to the bazaar and left my aspect wasn't gone. It just stayed there for the entire next stage, and afterwards i went to the bazaar again and nothing changed, and then right after leaving onto the next stage (stage 5), It just straight up gave me another aspect. 

Basically: Completely ignored removing the one it gave me earlier upon the second time entering the bazaar and ignored giving me a new one afterwards. I'm betting this may have caused a chain of confusion and just ended up giving me another aspect. (Note: after receiving the 2nd aspect, upon entering a new stage (stage 6) i got ANOTHER one. It just ignores replacing them altogether now.)

**Comment by ConqAra** [comment]
You may need to do a further check removing the given aspect upon leaving the stage, the artifact should probably ignore giving you the aspect altogether when in the bazaar.

**Comment by CuteDoge01** [comment]
Oh I see, I have fixed this in 1.0.4 but made a mistake so this was not working, my bad. Version 1.0.5 now has the right fixes, tested myself, works good now. Sorry for that, coding in 2 am be like, hah

It still gives aspect in all intermissions, just so now it correctly removes it when there is no teleporter

This is helpful when doing Golden Coast or Void Fields

I could make this so the aspect item is removed when the stage ends, but for some reason this makes Propersave mod (Which is very popular and cool) to duplicate aspects. No idea why. So I am doing it right now when the teleported event is over and making a fallback if the last level was intermission. Now it works fine, and Propersave does not duplicate aspects, great! It changes them, so you could abuse it to play with aspects you want but this is not a big deal.

Also sorry for the imbalanced run with triple aspects. If you have saved it via mod you can just scrap those items. Thank you so much for reporting it!!
